common/credentials.go

```diff
@@ -0,0 +1,84 @@
+package common
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/credentials/stscreds"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/spf13/viper"
+)
+
+type AwsCredentials struct {
+	ClientRegion   string
+	AwsAccessKeyId string
+	AwsSecretKey   string
+	SessionToken   string
+	AssumeRoleArn  string
+}
+
+func WithAwsCredentials() AwsCredentials {
+	return AwsCredentials{
+		ClientRegion:   viper.GetString("aws.clientRegion"),
+		AwsAccessKeyId: viper.GetString("aws.staticCredentials.awsAccessKeyId"),
+		AwsSecretKey:   viper.GetString("aws.staticCredentials.awsSecretKey"),
+		SessionToken:   viper.GetString("aws.staticCredentials.sessionToken"),
+		AssumeRoleArn:  viper.GetString("aws.assumeRoleArn"),
+	}
+}
+
+func (c AwsCredentials) credentialsProvider() (config.LoadOptionsFunc, error) {
+	var credProvider = config.WithCredentialsProvider(nil)
```

Comment:
This means default credential provider?

Reply:
yes, the result is default credential chain